contracts/src/deployers/savings-usds-l2/SavingsUSDSL2HyperdriveDeployerCoordinator.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.24;
+
+import { ERC20 } from "openzeppelin/token/ERC20/ERC20.sol";
+import { SafeERC20 } from "openzeppelin/token/ERC20/utils/SafeERC20.sol";
+import { SavingsUSDSL2Conversions } from "../../instances/savings-usds-l2/SavingsUSDSL2Conversions.sol";
+import { IHyperdrive } from "../../interfaces/IHyperdrive.sol";
+import { IPSM } from "../../interfaces/IPSM.sol";
+import { IHyperdriveDeployerCoordinator } from "../../interfaces/IHyperdriveDeployerCoordinator.sol";
+import { SAVINGS_USDS_L2_HYPERDRIVE_DEPLOYER_COORDINATOR_KIND } from "../../libraries/Constants.sol";
+import { ONE } from "../../libraries/FixedPointMath.sol";
+import { HyperdriveDeployerCoordinator } from "../HyperdriveDeployerCoordinator.sol";
+
+/// @author DELV
+/// @title SavingsUSDSL2HyperdriveDeployerCoordinator
+/// @notice The deployer coordinator for the SavingsUSDSL2Hyperdrive
+///         implementation.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+contract SavingsUSDSL2HyperdriveDeployerCoordinator is
+    HyperdriveDeployerCoordinator
+{
+    using SafeERC20 for ERC20;
+
+    /// @notice The deployer coordinator's kind.
+    string public constant override kind =
+        SAVINGS_USDS_L2_HYPERDRIVE_DEPLOYER_COORDINATOR_KIND;
+
+    /// @notice The PSM contract.
+    IPSM public immutable PSM;
```

Comment:
Why are we passing the PSM as extra data into the target deployers when we set it as an immutable here? Maybe I’m missing something, but we typically either use immutables everywhere or use extra data everything (including in the coordinator).

Reply:
Oh, great point, it is absolutely not necessary! I've switched to just using the extraData.